consensus/src/round_manager.rs

```diff
@@ -321,11 +322,26 @@ impl RoundManager {
             .proposer_election
             .is_valid_proposer(self.proposal_generator.author(), new_round_event.round)
         {
-            let proposal_msg =
-                ConsensusMsg::ProposalMsg(Box::new(self.generate_proposal(new_round_event).await?));
+            let proposal_msg = Box::new(self.generate_proposal(new_round_event).await?);
             let mut network = self.network.clone();
-            network.broadcast(proposal_msg).await;
-            counters::PROPOSALS_COUNT.inc();
+            if Self::should_inject_reconfiguration_error(proposal_msg.proposal().block_data()) {
+                // We send the proposal to half of the validators to force a timeout.
+                let mut half_peers: Vec<_> = self
+                    .epoch_state
+                    .verifier
+                    .get_ordered_account_addresses_iter()
+                    .collect();
+                half_peers.truncate(half_peers.len() / 2);
+                network
+                    .send(ConsensusMsg::ProposalMsg(proposal_msg), half_peers)
+                    .await;
+                return Err(anyhow::anyhow!("Injected error in reconfiguration suffix"));
+            } else {
```

Comment:
I kinda thought we wouldn't even be exposed to this kinda test code? I would have really liked to have seen this written in a way that it was obviously /not/ compiled in

Reply:
I wrote that way initially with fail_point! which doesn't support async. but with the current setup I should be able to mark this whole block with cfg. but probably another PR?

Reply:
Agree that this is pretty ugly.  Definitely should find a cleaner way to do it eventually.

Reply:
https://github.com/diem/diem/pull/9135#event-5282242042